apps/solver/test/equation_store.cpp

```diff
@@ -1,387 +1,185 @@
-  // x+𝐢 = 0 --> x = e^(-π/2×i)
-  const char * solutions0Polar[] = {"ℯ^\u0012-\u0012\u0012π\u0013/\u00122\u0013\u0013𝐢\u0013"}; // ℯ^(-(π/2)𝐢)
-  assert_equation_system_exact_solve_to(equations0,  EquationStore::Error::NoError, EquationStore::Type::LinearSystem, (const char **)variablesx, solutions0Polar, 1);
-
-  // x+√(-1) = 0 --> x = e^(-π/2𝐢)
-  assert_equation_system_exact_solve_to(equations1,  EquationStore::Error::NoError, EquationStore::Type::LinearSystem, (const char **)variablesx, solutions0Polar, 1);
-
-  // x^2+x+1=0
-  const char * solutions2Polar[] = {"ℯ^\u0012-\u0012\u00122π\u0013/\u00123\u0013\u0013𝐢\u0013","ℯ^\u0012\u0012\u00122π\u0013/\u00123\u0013\u0013𝐢\u0013", "3ℯ^\u0012π·𝐢\u0013"}; // ℯ^(-(2π/3)𝐢), ℯ^((2π/3)𝐢), 3ℯ^(π𝐢)
-  assert_equation_system_exact_solve_to(equations2, EquationStore::Error::NoError, EquationStore::Type::PolynomialMonovariable, (const char **)variablesx, solutions2Polar, 3);
-
-  // x^2-√(-1)=0
-  const char * solutions3Polar[] = {"ℯ^\u0012-\u0012\u00123π\u0013/\u00124\u0013\u0013𝐢\u0013", "ℯ^\u0012\u0012\u0012π\u0013/\u00124\u0013\u0013𝐢\u0013", "4ℯ^\u0012\u0012\u0012π\u0013/\u00122\u0013\u0013𝐢\u0013"}; // ℯ^(-(3×π/4)𝐢)", "ℯ^((π/4)𝐢)", "4ℯ^((π/2)𝐢)
-  assert_equation_system_exact_solve_to(equations3, EquationStore::Error::NoError, EquationStore::Type::PolynomialMonovariable, (const char **)variablesx, solutions3Polar, 3);
+QUIZ_CASE(equation_solve_complex_real) {
+  set_complex_format(Real);
+  assert_solves_to("x+𝐢=0", "x=-𝐢"); // We still want complex solutions if the input has some complex value
+  assert_solves_to_error("x+√(-1)=0", EquationUnreal);
+  assert_solves_to("x^2+x+1=0", {"delta=-3"});
+  assert_solves_to_error("x^2-√(-1)=0", EquationUnreal);
+  assert_solves_to_error("x+√(-1)×√(-1)=0", EquationUnreal);
+  assert_solves_to("root(-8,3)*x+3=0", "x=3/2");
+  reset_complex_format();
+}
 
-  const char * solutions5Polar[] = {"\u0012\u00123\u0013/\u00122\u0013\u0013ℯ^\u0012\u0012\u00122π\u0013/\u00123\u0013\u0013𝐢\u0013"}; //3/2ℯ^\u0012\u00122π\u0012/3\u0013𝐢"};
-  assert_equation_system_exact_solve_to(equations5,  EquationStore::Error::NoError, EquationStore::Type::LinearSystem, (const char **)variablesx, solutions5Polar, 1);
+QUIZ_CASE(equation_solve_complex_cartesian) {
+  set_complex_format(Cartesian);
+  assert_solves_to("x+𝐢=0", "x=-𝐢");
+  assert_solves_to("x+√(-1)=0", "x=-𝐢");
+  assert_solves_to({"x^2+x+1=0"}, {"x=-1/2-((√(3))/2)𝐢", "x=-1/2+((√(3))/2)𝐢", "delta=-3"});
+  assert_solves_to("x^2-√(-1)=0", {"x=-√(2)/2-(√(2)/2)𝐢", "x=√(2)/2+(√(2)/2)𝐢", "delta=4𝐢"});
+  assert_solves_to("x+√(-1)×√(-1)=0", "x=1");
+  assert_solves_to("root(-8,3)*x+3=0", "x=-3/4+(3√(3)/4)*𝐢");
+  reset_complex_format();
+}
 
-  // Put back the complex format
-  Poincare::Preferences::sharedPreferences()->setComplexFormat(Poincare::Preferences::ComplexFormat::Real);
+QUIZ_CASE(equation_solve_complex_polar) {
+  // TODO: Does it make sense to test polar vs cartesian?
```

Comment:
See with @EmilieNumworks, those tests are needed to make sure the complex format is taken into account